File: solutions/go/roman-numerals/1/roman_numerals.go
package romannumerals
import (
    "fmt"
    "strconv"
    "strings"
    "errors"
)

func ToRomanNumeral(input int) (string, error) {

    romansSymbols := []string{ "M", "D", "C", "L", "X", "V", "I"}
    symbolsNb := len(romansSymbols)
   
 	if input<=0 {
        return "", errors.New("Zero, or negative cannot be used")
    } else if input>=4000{
       return "", errors.New("4000 is out of range")
    }
	//code := strconv.Itoa(input)
	code := fmt.Sprintf("%04d", input)
    
    romanRes :=""
    for index, val := range code {
        subIndex := symbolsNb-(7-index*2)
     	integerValue, _ := strconv.Atoi(string(val)) 

        if(index==0){
        		romanRes += strings.Repeat(romansSymbols[subIndex], integerValue)
        }else if (index >=1 && index<=3){
			if integerValue>= 1 && integerValue <=3 {
            	romanRes += strings.Repeat(romansSymbols[subIndex], integerValue)
            } else if integerValue>= 5 && integerValue <=8 {
                romanRes+= romansSymbols[subIndex-1]
                romanRes+= strings.Repeat(romansSymbols[subIndex], integerValue%5)
            }else if integerValue==4 || integerValue==9 {
             	romanRes+= romansSymbols[subIndex]+romansSymbols[subIndex-(1+integerValue%4)]
                    
            }
        }
    }
    
	return romanRes, nil
}

File: solutions/javascript/roman-numerals/1/roman-numerals.js
//
// This is only a SKELETON file for the 'Roman Numerals' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const toRoman = (input) => {
  const romansSymbols = [ "M", "D", "C", "L", "X", "V", "I"]
  const symbolsNb = romansSymbols.length
  const code = [...String(input).padStart(4, '0')]
  let romanRes = ""
  
  for(let index=0; index < code.length; index++)
  {
    const subIndex = symbolsNb - (7-index*2)
    const currentVal = parseInt(code[index])

    if(index==0){
      romanRes += String(romansSymbols[subIndex]).repeat(currentVal)
      
    }else if(index >=1 && index<=3){
      
      if(currentVal>= 1 && currentVal <=3 ){
        
        romanRes += String(romansSymbols[subIndex]).repeat(currentVal)

      } else if(currentVal>= 5 && currentVal <=8 ){
        
        romanRes+= romansSymbols[subIndex-1]
        romanRes+= String(romansSymbols[subIndex]).repeat(currentVal%5)
        
      }else if(currentVal==4 || currentVal==9 ){
        
        romanRes+= romansSymbols[subIndex]+romansSymbols[subIndex-(1+currentVal%4)]
      }
    }
  }
  return romanRes
};